src/compas/artists/artist.py

```diff
@@ -27,35 +27,73 @@ def redraw():
 
 @pluggable(category="factories", selector="collect_all")
 def register_artists():
+    """Registers artists available in the current context."""
     raise NotImplementedError
 
 
-def identify_context():
+def is_viewer_open():
+    """Returns True if an instance of the compas_view2 App is available.
+
+    Returns
+    -------
+    bool
+
+    """
+    # TODO: implement [without introducing compas_view2 as a dependency..?]
+    return False
+
+
+def is_plotter_open():
+    """Returns True if an instance of the Plotter is available.
+
+    Returns
+    -------
+    bool
+
+    """
+    # TODO: implement
+    return False
+
+
+def _choose_context():
+    """Chooses an appropriate context depending on available contexts and open instances. with the following priority:
+    1. Viewer
+    2. Plotter
+    3. Rhino / GH
+    4. Other - randome choice from Artist.ITEM_ARTISTS.values()
```

Comment:
should this not just be combined into "CAD"? since you can only be in one CAD environment at the same time...